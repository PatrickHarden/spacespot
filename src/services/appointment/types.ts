export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
