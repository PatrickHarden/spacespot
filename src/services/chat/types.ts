export interface ChatAction {
  type: string
  payload?: ChatResponse | ChatMessage | string
}
export interface MessageAction {
  type: string
  payload?: string
}
export interface ChatState {
  data?: ChatResponse
  error?: string
}

export interface ChatResponse {
  chatId: number
  enquiryId: number
  peer1Id: string
  peer2Id: string
  messages: Array<ChatMessage>
}

export interface ChatMessage {
  userId: string
  message: string
  sentAt: string
}
