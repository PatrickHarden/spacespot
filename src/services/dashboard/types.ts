import { Space } from 'services/space/types'

export interface DashboardState {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
  error?: string
  isLoading: boolean
  initOK: boolean
  pendingDelete: Array<string>
}
