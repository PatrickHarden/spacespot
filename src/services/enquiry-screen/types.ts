export interface EnquiryScreenAction {
  type: string
  payload: EnquiryScreenStateData | string
}
export interface EnquiryScreenState {
  data?: EnquiryScreenStateData
  error?: string
}
export interface EnquiryScreenStateData {
  id: string
}
