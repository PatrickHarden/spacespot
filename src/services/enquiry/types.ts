import { FileItem } from 'services/global/types'
import { NegotiationTerm } from 'services/negotiation/types'

export interface DocumentDTO {
  fileId: number
  fileName: string
  enquireId: number
  uploadAt: Date
  uploadedBy: string
  documentType?: string
}

export const DOC_TYPE_TEMPLATE = 'CompanyTemplate'

export type DocData = any

export interface EnquiryAction {
  type: string
  payload:
    | string
    | number
    | DocumentDTO
    | Array<DocumentDTO>
    | EnquiryResponse
    | FileItem
    | DocData
    | EnquiryRequest
}

export interface EnquiryState {
  data?: EnquiryResponse
  docs?: Array<DocumentDTO>
  error?: string
}

export enum EnquiryType {
  HotDesk = 'HotDesk',
  FixedDesk = 'FixedDesk',
  ServicedOffice = 'ServicedOffice',
  notChecked = '',
}

export interface EnquiryResponse {
  chatId: number
  enquireId: number
  spaceId: string
  /* Tenant */
  peer1Id: string
  /* Landlord */
  peer2Id: string
  negotiationId?: number
  enquireStatus?: string
  flexType?: EnquiryType
  flex: boolean
}

export interface EnquiryAPIResultActions {
  successSignInAfter?: boolean
  errorShowMessage?: string
}

export interface EnquiryRequest {
  spaceId: string
  message: string
  negotiation: {
    flexible: boolean
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}

export interface EnquiryRequestFlex {
  spaceId: string
  message: string
  flex: boolean
  flexType: EnquiryType
  flexIdentifier?: string
  negotiation: {
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}
