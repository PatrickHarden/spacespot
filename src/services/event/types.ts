export interface EnquiryEvent {
  eventId: number
  eventMessage: any
  eventType: string
  eventDate: Date
}

export interface EnquiryEventState {
  [enquiryId: string]: {
    events: Array<EnquiryEvent>
    isLoading?: boolean
    error?: string
  }
}

export interface GetEnquiryEventsAction {
  type: string
  payload: {
    enquiryId: string
  }
}

export interface GetEnquiryEventsSuccessAction {
  type: string
  payload: {
    enquiryId: string
    events: Array<EnquiryEvent>
  }
}

export interface GetEnquiryEventsErrorAction {
  type: string
  payload: {
    enquiryId: string
    error: string
  }
}

export type EnquiryEventAction =
  | GetEnquiryEventsAction
  | GetEnquiryEventsSuccessAction
  | GetEnquiryEventsErrorAction
