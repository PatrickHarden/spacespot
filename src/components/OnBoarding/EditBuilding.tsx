import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useIntl } from 'react-intl'
import styled from 'styled-components'
import { useParams } from 'react-router-dom'

import dashboardSelectors from 'services/dashboard/selectors'
import dashboardActions from 'services/dashboard/actions'
import onboardingSelectors from 'services/onboarding/selectors'
import onboardingActions from 'services/onboarding/actions'
import BuildingForm from './BuildingForm'
import Spinner from 'components/icons/Spinner'
import Container from './MainContainer'

const Title = styled.div`
  padding: 40px 0 0 0;
  color: #404042;
  font-size: 32px;
  font-weight: 500;
  line-height: 41px;
`

const SpinnerContainer = styled.div`
  text-align: center;
  margin-top: 20px;
`

const EditBuilding = () => {
  const { buildingId } = useParams()
  const id = buildingId ? buildingId : '0'
  const dispatch = useDispatch()
  const initOK = useSelector(dashboardSelectors.isDashboardInitOK)
  const loading = useSelector(dashboardSelectors.isDashboardLoading)
  const building = useSelector(
    dashboardSelectors.selectDashboardBuildingCurried,
  )(id)
  const onboardingState = useSelector(onboardingSelectors.onboardingState)
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  useEffect(() => {
    if (!loading) {
      if (!initOK) {
        dispatch(dashboardActions.dashboardInit())
      } else if (building) {
        dispatch(onboardingActions.onboardingEditBuilding(building))
      }
    }
  }, [building, dispatch, initOK, loading])

  return (
    <Container>
      <Title>{t('ONBOARDING_TITLE_EDIT_BUILDING')}</Title>
      {loading || buildingId !== onboardingState.buildingId ? (
        <SpinnerContainer>
          <Spinner />
        </SpinnerContainer>
      ) : (
        <BuildingForm isEditing={true} />
      )}
    </Container>
  )
}
export default EditBuilding
