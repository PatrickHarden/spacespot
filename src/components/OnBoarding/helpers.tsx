import {
  OnBoardingState,
  SpaceData,
  SpaceStatus,
  AvailabilityFlexPriceInterval,
} from 'services/onboarding/types'

const date = new Date('2020/01/01')

export const createSpace = (type: string): SpaceData => ({
  id: '0',
  type,
  spaceName: '',
  spaceDescription: '',
  spaceHighlights: '',
  spaceSize: undefined,
  spaceFloor: undefined,
  spaceSizeCommon: undefined,
  spaceMatterPort: '',
  spaceFloored: '',
  use: '',
  spaceRent: undefined,
  spaceServices: undefined,
  spaceServicesNotNegotiable: false,
  availabilityFixed: date,
  months: undefined,
  hotDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  fixedDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  servicedOffices: [],
  images: [],
  floorPlan: [],
  fitout: [{ description: 'FitOut1', name: 'FitOut1', amount: 1000 }],
  status: SpaceStatus.ReadyForPublish,
})

export const createOnBoardingState = (): OnBoardingState => ({
  isNew: true,
  buildingId: undefined,
  images: [
    {
      key: 0,
      name: '40VilliersStreet.External_Photo_1_small.png',
      type: 'image/png',
      uri:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==',
    },
  ],
  docs: [],
  name: '',
  description: '',
  address: '',
  city: '',
  postCode: '',
  country: '',
  location: '',
  latLng: undefined,
  district: '',
  subdistrict: {
    id: 0,
    name: '',
  },
  amenities: { hasGym: { checked: true, desc: 'Gym' } },
  spaces: { '0': createSpace('FLEX') },
  spaceCount: 1,
})

export const valInteger = (
  current: number | undefined,
  value: number,
  min: number,
  max?: number,
): number => {
  const val = Number(Number(value).toFixed())
  return val < min || (max && val > max) ? current || 0 : val
}
