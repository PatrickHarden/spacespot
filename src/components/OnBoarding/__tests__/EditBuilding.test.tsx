import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import { SpacespotState } from 'services/global/types'
import EditBuilding from '../EditBuilding'
import { createOnBoardingState } from '../helpers'
import msgs from 'translations/en.json'

const mockState: any = {
  onboarding: createOnBoardingState(),
  dashboard: {
    isLoading: true,
    buildings: [
      {
        'Common.PrimaryKey': '0',
      },
    ],
  },
}

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    buildingId: '0',
  }),
}))

jest.mock('react-redux', () => ({
  useSelector: (selector: (state: SpacespotState) => any) =>
    selector(mockState),
  useDispatch: () => {
    return jest.fn()
  },
}))

jest.mock('@react-google-maps/api', () => {
  const GoogleMap = () => <div />
  const Marker = () => <div />
  const Autocomplete = () => <div />
  const useLoadScript = () => ({ isLoaded: true })
  return { GoogleMap, Marker, Autocomplete, useLoadScript }
})

it('renders Spinner ', async () => {
  const wrapper = render(
    <IntlProvider locale="en-GB" messages={msgs}>
      <EditBuilding />
    </IntlProvider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('renders EditBuilding ', async () => {
  mockState.dashboard.isLoading = false
  const wrapper = render(
    <IntlProvider locale="en-GB" messages={msgs}>
      <EditBuilding />
    </IntlProvider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('loads dashboard', async () => {
  mockState.dashboard = {}
  const wrapper = render(
    <IntlProvider locale="en-GB" messages={msgs}>
      <EditBuilding />
    </IntlProvider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})
