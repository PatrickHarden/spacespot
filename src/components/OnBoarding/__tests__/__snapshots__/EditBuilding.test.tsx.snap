// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads dashboard 1`] = `
<DocumentFragment>
  <div
    class="sc-cvbbAY dBmUDY"
  >
    <div
      class="sc-jWBwVP gcMrf"
    >
      Edit a building
    </div>
    <div
      class="sc-brqgnP dosQox"
    >
      <div
        class="sc-eHgmQL dvFuqY"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders EditBuilding  1`] = `
<DocumentFragment>
  <div
    class="sc-cvbbAY dBmUDY"
  >
    <div
      class="sc-jWBwVP gcMrf"
    >
      Edit a building
    </div>
    <div
      class="sc-brqgnP dosQox"
    >
      <div
        class="sc-eHgmQL dvFuqY"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Spinner  1`] = `
<DocumentFragment>
  <div
    class="sc-cvbbAY dBmUDY"
  >
    <div
      class="sc-jWBwVP gcMrf"
    >
      Edit a building
    </div>
    <div
      class="sc-brqgnP dosQox"
    >
      <div
        class="sc-eHgmQL dvFuqY"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
